sdk/identity/Azure.Identity/src/AzureCliCredentialClient.cs

```diff
@@ -0,0 +1,197 @@
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                CreateNoWindow = true,
+                WorkingDirectory = _workingDir,
+                Environment = { { "PATH", _path } }
+            };
+
+            return await RunProcessAsync(async, procStartInfo, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async ValueTask<(string, int)> RunProcessAsync(bool async, ProcessStartInfo procStartInfo, CancellationToken cancellationToken)
+        {
+            int exitCode = default;
+            StringBuilder stdOutput = new StringBuilder();
+            StringBuilder stdError = new StringBuilder();
+            bool completed;
+
+            using (Process proc = new Process())
+            {
+
+                proc.StartInfo = procStartInfo;
+                proc.OutputDataReceived += new DataReceivedEventHandler((sender, e) => stdOutput.AppendLine(e.Data));
+                proc.ErrorDataReceived += new DataReceivedEventHandler((sender, e) => stdError.AppendLine(e.Data));
+
+                proc.Start();
+                proc.BeginOutputReadLine();
+                proc.BeginErrorReadLine();
+
+                completed = async ? await Task.Run(() => proc.WaitForExit(CliProcessTImeoutMs), cancellationToken).ConfigureAwait(false) : proc.WaitForExit(CliProcessTImeoutMs);
+
+                if (!completed)
+                {
+                    proc.CancelOutputRead();
+
+                    proc.CancelErrorRead();
+
+                    throw new AuthenticationFailedException(AzureCliTimeoutError);
+                }
+
+                exitCode = proc.ExitCode;
+            }
+
+            if (exitCode != 0)
+            {
+                return (stdError.ToString(), exitCode);
+            }
+            else
+            {
+                return (stdOutput.ToString(), exitCode);
+            }
+        }
+
+        private static AccessToken DeserializeOutput(string output)
+        {
+            string accessToken = null;
+            DateTimeOffset expiresOn = DateTimeOffset.MaxValue;
+
+            using (JsonDocument json = JsonDocument.Parse(output))
+            {
+
+                foreach (JsonProperty prop in json.RootElement.EnumerateObject())
+                {
+                    switch (prop.Name)
+                    {
+                        case "accessToken":
+                            accessToken = prop.Value.GetString();
+                            break;
+
+                        case "expiresIn":
+                            expiresOn = DateTimeOffset.UtcNow + TimeSpan.FromSeconds(prop.Value.GetInt64());
+                            break;
+
+                        case "expiresOn":
+                            if (expiresOn == DateTimeOffset.MaxValue)
+                            {
+                                var expiresOnStr = prop.Value.GetString();
```

Comment:
This can be "N/A"

Reply:
I'm not handling this specifically. If expiresOn is "N/A" then ParseExact will throw a FormatException, this will be caught and turned into an AuthenticationFailed exception stating that the AzureCliCredential failed for this reason.